test/shewanella_kmer1_kmer2.pl

```diff
@@ -23,206 +30,80 @@
              "call_features_CDS_glimmer3"=>'0',
-             "call_features_CDS_prodigal"=>'0',
+             "call_features_CDS_prodigal"=>'1',
              "annotate_proteins_kmer_v2"=>'1',
              "kmer_v1_parameters"=>'1',
-             "annotate_proteins_similarity"=>'1',
+             "annotate_proteins_similarity"=>'0',
              "retain_old_anno_for_hypotheticals"=>'0',
-             "resolve_overlapping_features"=>'0',
+             "resolve_overlapping_features"=>'1',
              "call_features_prophage_phispy"=>'0',
              "output_genome"=>$genome_obj_name,
              "workspace"=>get_ws_name()
            };
     return make_impl_call("RAST_SDK.annotate_genome", $params);
 }
 
-my $diff_count = 0;
-my $num_func = 0;
+my $diff_count   = 0;
+my $num_func_in  = 0;
 my $num_func_out = 0;
+my $genome_obj_name = "Shewanella_sp._ANA-3";
 lives_ok {
-    my $genome_obj_name = "genome.1";
     my $genome_gbff_name = "Shewanella_sp._ANA-3.gbff";
     my ($tmp_genome_obj, $genome_ref) = prepare_gbff($genome_gbff_name,$genome_obj_name);
-	my $orig_genome = $ws_client->get_objects([{ref=>$genome_ref}])->[0]->{data};
-
-    if (defined($orig_genome->{taxon_ref})) {
-        $orig_genome->{taxon_ref} = "ReferenceTaxons/unknown_taxon" ;
-    }
-    if (defined($orig_genome->{genbank_handle_ref})) {
-        delete $orig_genome->{genbank_handle_ref};
-    }
-
-	my $orig_funcs = {};
-	my %types;
-	for (my $i=0; $i < scalar @{$orig_genome->{features}}; $i++) {
-		my $ftr = $orig_genome->{features}->[$i];
-		my $func = $ftr->{function};
-		if (not defined($func)) {
-			$func = "";
-		}
-		if (not defined($ftr->{type})) {
-			$ftr->{type} = "other";
-		}
-		if ($ftr->{type} eq "gene" and not(defined($ftr->{protein_translation}))) {
-			if (lc($ftr->{function}) =~ /ribosomal/)
-			{
-				$ftr->{type} = 'rRNA';
-			}
-			elsif (lc($ftr->{function}) =~ /trna/)
-			{
-				$ftr->{type} = 'tRNA';
-			}
-			else
-			{
-				$ftr->{type} = 'other non-coding';
-			}
-		}
-		if (exists $types{$ftr->{type}})
-		{
-			$types{$ftr->{type}} ++;
-		}
-		else
-		{
-			$types{$ftr->{type}} = 1;
-		}
-		$num_func++;
-        $orig_funcs->{$ftr->{id}} = $func;
-    }
+#	my $genome_ref = "15792/209997/1";
 
-#	print "**** Number of input features = $num_func\n";
-#	foreach my $key (keys %types)
-#	{
-#		print "\t$key = $types{$key}\n";
-#	}
+	my ($orig_genome,$orig_funcs) = &get_and_prep($genome_ref);
 
-    print("Running RAST annotation\n");
-    my $ret = reannotate_genome($genome_obj_name, $genome_ref);
-	my $report_ref = $ret->{report_ref};
-    my $report_obj = $ws_client->get_objects([{ref=>$report_ref}])->[0]->{data};
-	my $report_text = $report_obj->{direct_html};
-    print "\nReport: " . $report_text . "\n\n";
-    my $genome_ref = get_ws_name() . "/" . $genome_obj_name ;
-    my $genome_obj = $ws_client->get_objects([{ref=>$genome_ref}])->[0]->{data};
+	print "number of input features = ".scalar  @{$orig_genome->{features}}."\n";
```

Comment:
Asserting an expected value for results in tests would also make them more helpful.

Reply:
Most of the print statements were to help me verify that I was making the right changes in the Impl file. I added as assert for this genome because changing the annotation was the purpose of the selected options.